genai-perf/genai_perf/export_data/telemetry_data_exporter_util.py

```diff
@@ -0,0 +1,190 @@
+        "p50",
+        "p25",
+    ]
+
+    TELEMETRY_CONSTANT_METRICS_HEADER = [
+        "Metric",
+        "GPU",
+        "Value",
+    ]
+
+    TELEMETRY_CONSTANT_METRICS = ["gpu_power_limit", "total_gpu_memory"]
+
+    TELEMETRY_GROUPS = {
+        "Power": ["gpu_power_usage", "gpu_power_limit", "energy_consumption"],
+        "Memory": ["gpu_memory_used", "total_gpu_memory"],
+        "Utilization": ["gpu_utilization"],
+    }
+
+    @staticmethod
+    def merge_telemetry_stats_json(
+        telemetry_stats: Optional[Dict], stats_and_args: Dict
+    ) -> None:
+        if telemetry_stats is not None:
+            stats_and_args.update({"telemetry_stats": telemetry_stats})
+
+    @staticmethod
+    def export_telemetry_stats_csv(telemetry_stats: Optional[Dict], csv_writer) -> None:
+        if telemetry_stats:
+            TelemetryDataExporterUtil._write_dynamic_telemetry_stats(
+                telemetry_stats, csv_writer
+            )
+            TelemetryDataExporterUtil._write_constant_telemetry_stats(
+                telemetry_stats, csv_writer
+            )
+
+    @staticmethod
+    def export_telemetry_stats_console(
+        telemetry_stats: Optional[Dict], STAT_COLUMN_KEYS: List[str], console: Console
+    ) -> None:
+        if telemetry_stats:
+            TelemetryDataExporterUtil._construct_telemetry_stats_table(
+                telemetry_stats, STAT_COLUMN_KEYS, console
+            )
+
+    @classmethod
+    def _construct_telemetry_stats_table(
+        cls, telemetry_stats, STAT_COLUMN_KEYS: List[str], console: Console
+    ) -> None:
+        for group_name, metrics in cls.TELEMETRY_GROUPS.items():
+            table = Table(title=f"NVIDIA GenAI-Perf | {group_name} Metrics")
+
+            for metric_name in metrics:
+                metric_data = telemetry_stats.get(metric_name, {})
+
+                unit = metric_data.get("unit", "N/A")
+                metric_name_display = cls._format_metric_name(
+                    metric_name.replace("_", " ")
+                )
+                table_title = f"{metric_name_display}{f' ({unit})' if unit else ''}"
+                sub_table = Table(title=table_title)
+
+                sub_table.add_column(
+                    "GPU Index", justify="right", style="cyan", no_wrap=True
+                )
+                for stat in STAT_COLUMN_KEYS:
+                    sub_table.add_column(stat, justify="right", style="green")
+
+                cls._construct_telemetry_stats_subtable(
+                    sub_table, metric_data, metric_name, STAT_COLUMN_KEYS
+                )
+                table.add_row(sub_table)
+
+            console.print(table)
+
+    @classmethod
+    def _construct_telemetry_stats_subtable(
+        cls,
+        table: Table,
+        metric_data: Dict[str, Dict[str, float]],
+        metric_name: str,
+        STAT_COLUMN_KEYS: List[str],
+    ) -> None:
+        gpu_indices = sorted([key for key in metric_data.keys() if key != "unit"])
```

Comment:
Can you review this function? You're sorting and then using a nested loop to get the values in this dictionary. I wonder if there's a more efficient way to do this.